registry/storage/driver/s3/s3.go

```diff
@@ -287,8 +294,8 @@ func (d *driver) WriteStream(path string, offset int64, reader io.Reader) (total
 		return 0, err
 	}
 
-	buf := make([]byte, d.ChunkSize)
-	zeroBuf := make([]byte, d.ChunkSize)
+	buf := d.getbuf()
+	zeroBuf := d.getbuf()
```

Comment:
Since zeroBuf is only used when offset >0, can this getBuf and putBuf just be put at the boundaries of that block?